common-template/src/main/java/com/redhat/cloud/notifications/qute/templates/TemplateService.java

```diff
@@ -166,4 +166,13 @@ public Map<String, Object> convertActionToContextMap(final Action action) {
         return objectMapper
             .convertValue(action.getContext(), new TypeReference<Map<String, Object>>() { });
     }
+
+    public boolean isValidTemplateDefinition(final TemplateDefinition config) {
+        try {
+            compileTemplate(config);
+        } catch (TemplateNotFoundException e) {
```

Comment:
What if the template exists but doesn't compile for some reason? Does that make the template definition valid? Or should we catch additional exceptions here or an exception with a larger scope?